autotest/ogr/ogr_rfc35.py

```diff
@@ -449,23 +504,25 @@ def test_ogr_rfc35_mitab_4():
     assert feat.GetField("oldintfld") == "98765"
     feat = None
 
-    CheckFeatures(lyr, field3="baz5")
+    CheckFeatures(ds, lyr)
 
 
 ###############################################################################
 # Test DeleteField()
 
 
-def test_ogr_rfc35_mitab_5():
+def test_ogr_rfc35_5(rfc35_test_input, driver_name, tmp_path):
+
+    if driver_name == "Memory":
+        ds = rfc35_test_input
+    else:
+        ds = ogr.Open(rfc35_test_input, update=1)
 
-    ds = ogr.Open("tmp/rfc35_test.tab", update=1)
     lyr = ds.GetLayer(0)
     lyr_defn = lyr.GetLayerDefn()
 
     assert lyr.TestCapability(ogr.OLCDeleteField) == 1
 
-    assert lyr.DeleteField(0) == 0
```

Comment:
is this removal intentional ?

Reply:
Yes - in `master` the tests in this file run sequentially, so this line is cleaning up the addition of `oldintfld` near the end of the previous test. In this PR, each test gets a clean copy of the input dataset.